File: Lab2/Hailstone.py
x = int(input('Please give an integer: '))
step = 0
print(x, end=' ')
while x!= 1 :
    step += 1
    if x % 2 == 0 :
        x //= 2
    else: 
        x = x*3 +1
    print(' -> ',x,end=' ')
else:
    print('\nTotal number of steps: ', step)

File: Lab2/Hailstone.cpp
#include <iostream>
int main(){
    int x;
    int step = 0;
    std::cout << "Please give a number: ";
    std::cin >> x;

    std::cout << x;
    while(x != 1){
        step++;
        if(x % 2 == 0) x/=2;
        else x = x*3 + 1;
        std::cout << " -> " << x;
    }

    std::cout << "\nTotal number of steps: " << step << std::endl;

}